tests/unit/test_charm.py

```diff
@@ -19,6 +19,19 @@
 )
 
 
+def infinite_time_values(start=0, step=2):
+    """Generator that returns an infinite sequence of time values.
+
+    Args:
+        start: Initial time value.
+        step: Time step between values.
+    """
+    current_time = start
+    while True:
+        yield current_time
+        current_time += step
```

Comment:
You could use `itertools.count()` for this

```
>>> from itertools import count
>>> a = count(0,2)
>>> next(a)
0
>>> next(a)
2
>>> next(a)
4
>>> next(a)
6
>>> next(a)
8
```

Reply:
Nice ✅